### PR TITLE
23

### PR DESCRIPTION
**Pull request by praveen-betalectic**

### CODE REVIEW

File: src/repositories/neptune.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+const prepareUserContactInfoPayload = async (payload) => {
+	let neptuneData = {};
+
+	switch (payload.type) {
+		case "email":
+			neptuneData = {
+				type: "email",
+				value: payload.value,
+				filterable: "regular",
```
Comment (rjvim) [inline src/repositories/neptune.js:18]: Do we need this "regular" here? I think we don't need this.

**Review by rjvim — CHANGES_REQUESTED** [review]
Left a comment @praveen-betalectic

File: src/repositories/neptune.js (hunk 1 above)
Comment (praveen-betalectic) [reply src/repositories/neptune.js:18]: i followed Neptune's documentation there.

Comment (praveen-betalectic) [reply src/repositories/neptune.js]: @rjvim made changes please review now.

**Comment by rjvim** [comment]
@praveen-betalectic I just want to confirm, what happens if there no NEPTUNE related variables are not present in .env.

nodejs SDK for Neptune handles those?

**Comment by rjvim** [comment]
1. Check for Neptune ENV Variables in nodejs package itself
2. Move neptune.fire function to neptune.js file
3. Add try/catch, but don't throw error if anything failed in Neptune SDK side

**Comment by praveen-betalectic** [comment]
@rjvim as we are firing many events from Neptune 
- if we can write a function to check for the env variables 
- which will return true if both the variables are there `NEPTUNE_ENV` and `NEPTUNE_TOKEN`
 ```
const envSet = () => {
	if (process.env.NEPTUNE_ENV && process.env.NEPTUNE_TOKEN) {
		return true;
	}
	return false;
};
```
- we can trigger that function before every event,
- if the function return `true` then we can proceed with the event 
- else we can skip the event and  continue with the rest of the code 

is this approach acceptable?

**Comment by rjvim** [comment]
This sounds great @praveen-betalectic. Go ahead.

**Comment by praveen-betalectic** [comment]
@rjvim made the changes as you instructed, please review now.

**Comment by rjvim** [comment]
@praveen-betalectic Is the PR pending merging, or we need to close it?

**Comment by praveen-betalectic** [comment]
@rjvim yes it is not merged yet,  we need to merge it.

**Pull request merged**